### DISCUSSION TITLE
Encode Obsidian settings in variables

### DISCUSSION THREAD

**Discussion by amzon-ex** [Taitava/obsidian-shellcommands#321] [Ideas]
Hi, before putting forward my feature request, I wish to thank you for making such an excellent plugin.

My request is to be able to encode the values of Obsidian settings in predefined variables: accent colour value or fonts, for instance. The reason why I ask this is it would make it possible to pass on these values to a command or write a conditional statement based on them, wherever required. Is it possible to achieve this?

Also, if there is a comprehensive way to do this without incorporating it explicitly into the plugin, please let me know.

**Comment by Taitava** [comment]
Hi! Thank you for your nice feedback and request! ❤️ 

The idea is good. I need to do some planning on this:
 - How well is settings retrieval supported by the public API of Obsidian? If Iremember correctly, there is no public support to get **all** settings. But at least private API access is possible by using `this.app.vault.getConfig("attachmentFolderPath")` ([source](https://forum.obsidian.md/t/api-get-the-directory-of-the-default-location-for-new-attachments-setting/36847/2?u=jare)), but it might break during Obsidian version updates.
     - Need to explore what settings are available via that particular private API method. What other methods could be used for getting other settings?
 - Decide a variable name. One candidate: `{{obsidian_setting:settingGroup.settingName}}`
     - So I'm thinking about creating a single variable that would cover all settings.
 - Develop gradually? Start with e.g. _Editor_ settings, and then proceed to adding _Files & Links_ etc settings later? At least if getting different settings groups requires using different API methods.

> The reason why I ask this is it would make it possible to pass on these values to a command or write a conditional statement based on them, wherever required.

Do you have some practical usage example for this? 🙂What kind of conditional statements would you make based on appearance settings (such as fonts/colors)? I do understand the general need for this kind of feature, but knowing realworld use cases can still improve planning the feature. 👍 

> Also, if there is a comprehensive way to do this without incorporating it explicitly into the plugin, please let me know.

In theory, you could parse the Obsidian settings files (e.g. `.obsidian/appearance.json` for one) in your shell command and retrieve the values you want.

**Comment by amzon-ex** [comment]
Hi @Taitava , thanks for taking notice!

I don't think I have the expertise to assist in the planning phase, but I can answer the question you asked:

> Do you have some practical usage example for this? 🙂What kind of conditional statements would you make based on appearance settings (such as fonts/colors)? I do understand the general need for this kind of feature, but knowing realworld use cases can still improve planning the feature. 👍

The reason why I thought having such a feature would be nice is because I compile markdown files with pandoc to obtain TeX/PDF files. I thought it would be great if I could pass on the fonts I use in the editor, or the accent colour to format the output file, but for that I'd need to get that info somehow. Does that help?

> In theory, you could parse the Obsidian settings files (e.g. .obsidian/appearance.json for one) in your shell command and retrieve the values you want.

This sounds useful, but also seems like it would increase the runtime of the command significantly given my use-case is frequent.

**Reply by Taitava** [reply]
> 
> The reason why I thought having such a feature would be nice is because I compile markdown files with pandoc to obtain TeX/PDF files. I thought it would be great if I could pass on the fonts I use in the editor, or the accent colour to format the output file, but for that I'd need to get that info somehow. Does that help?

Thank you for this example! 🙂 Yes, it definitely helps, and makes sense. 👍 
> 
> > In theory, you could parse the Obsidian settings files (e.g. .obsidian/appearance.json for one) in your shell command and retrieve the values you want.
> 
> This sounds useful, but also seems like it would increase the runtime of the command significantly given my use-case is frequent.

I don't think it will increase the runtime. The settings files are not so big after all (at least my Obsidian settings files are just a few tens of lines each). But anyway, doing the parsing in the shell command might be cumbersome and leads the focus a bit off from the main point - to compile markdown files with pandoc.

I'll keep this feature in mind, but can't promise it soon. I need to prioritize other, more commonly needed features at this point. But this is still something I'd like to have at some point. And if there are more people wanting this too out there, please leave a comment below. 🙂
